### ISSUE TITLE
Safari does not read float pixels 

### ISSUE THREAD

**Issue by dy** [regl-project/regl#487]
That results in broken `read` test, related to #479.

ThreeJS demo addressing the issue https://threejs.org/examples/webgl_read_float_buffer.html

**Comment by rreusser** [comment]
Pro: it's really hard to track down all the vendor inconsistencies so that I'd assume most of these corner cases simply remain broken unless the library accounts for them or unless your particular project actually receives enough testing that someone notices it and actually complains—which probably 2% of projects do.

Con: you can perfectly well accomplish this without building it into the library itself.

Philosophically, I'm definitely against it. But realistically, maybe it *is* the sort of thing that a library should account for.

**Comment by dy** [comment]
Mb we could just display an error/warning in case if read float is not supported and user attempts to do so, instead of silent failure.

**Comment by rreusser** [comment]
Maybe a good candidate for working this into `check` somehow so that we can issue a warning only in the non-production version?

**Comment by rreusser** [comment]
So I guess that'd make the behavior: `regl.read(new Float32Aray(...))` issues: `Warning: Reading floating point values is not permitted in some browsers. For a fallback option, please see: http://github.com/...`

**Comment by rreusser** [comment]
FWIW I'm mostly good on catching `pow(negativeNumber, ...)` in glsl these days, but for some reason I've been having lots of issues with medium precision not being adequate lately. Only a tangential discussion, but maybe there's a larger place in regl for warning about particularly vendor-dependent quirks and bugs so that you can at least minimize what's pretty much just an expectation that your webgl stuff will fail for one reason or another on a not-insignificant fraction of platforms you haven't tested it on.

**Comment by dy** [comment]
Not sure if everyone needs to see this warning in console, even in dev version - usually browsers do not care about alternative [broken] implementations.
But if the use-case happened in Safari it definitely should display a warning on `regl.read(floatArray)`. That is the Firefox practice for many features, eg. https://github.com/regl-project/regl/issues/488

**Comment by rreusser** [comment]
Agreed. Maybe I just need to write down a WebGL Compatibility Checklist.

**Comment by dy** [comment]
> a larger place in regl for warning about particularly vendor-dependent quirks and bugs 

That seems to be the case for IE, since it does not support a lot of glsl features, like `smoothstep`, sometimes there are even non-predictable quirks like unability to read texture etc. But not sure that's worth it for now - I'd at least collect 3-4 more precedents.

**Comment by rreusser** [comment]
Let's see… mobile browsers support floating point textures, but not writing to floating point framebuffers… oes_texture_float is available but you have to feature-detect rendering to it, I think… that one always gets me…

**Comment by dy** [comment]
Fixed in https://github.com/regl-project/regl/pull/479/commits/6afda7ca1b663835dcec0b11bce2c7e2eeab60a1. Should check in mobile browser too.
